src/QCheck.ml

```diff
@@ -783,11 +795,17 @@ module Test = struct
     | CR_continue
     | CR_yield of 'a TestResult.t
 
+  (* Get the name of a test *)
+  let name_ cell = match cell.name with
```

Comment:
Not your fault, but maybe this is the right opportunity to have `cell.name: string` and generate fresh names at cell construction (so that at least, anonymous tests are named "<test1>", "<test2>", …).